units/patch_intent.go

```diff
@@ -535,6 +526,69 @@ func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patc
 	return previousPatch.Status, nil
 }
 
+// updatePatchDocPatches updates the commit sha for existing patch modules and adds new modules to the patch for modules
+// that exist in the project but not in the patch.  This is necessary so that the manifest load endpoint will include the
+// updated commit sha in the response's module overrides field.
+func updatePatchDocPatches(ctx context.Context, patchDoc *patch.Patch, project *model.Project, token string) error {
+	var err error
+	for i := range patchDoc.Patches {
+		if patchDoc.Patches[i].ModuleName != "" {
+			// validate the module exists
+			var module *model.Module
+			module, err = project.GetModuleByName(patchDoc.Patches[i].ModuleName)
+			if err != nil {
+				return errors.Wrapf(err, "finding module '%s'", patchDoc.Patches[i].ModuleName)
+			}
+			if module == nil {
+				return errors.Errorf("module '%s' not found", patchDoc.Patches[i].ModuleName)
+			}
+			branch, err := getBranch(ctx, module.Repo, patchDoc.Patches[i].ModuleName, module.Branch, token)
+			if err != nil {
+				return errors.Wrapf(err, "getting branch for module '%s'", patchDoc.Patches[i].ModuleName)
+			}
+			patchDoc.Patches[i].Githash = *branch.Commit.SHA
+		}
+	}
+	for _, mod := range project.Modules {
```

Comment:
Could this just be the main loop, and in the if case at 556 we do the work above, otherwise we do 563? (Seems like  we're doing duplicate work, since above we only continue to getBranch if module exists in the project)

Reply:
Done